### PR TITLE
docs(docs): Initial docset & expanded README

### PR DESCRIPTION
**Pull request by ttomsu**

### CODE REVIEW

**Comment by ttomsu** [comment]
Friendly ping

File: README.md [hunk 1]
```diff
@@ -1,2 +1,96 @@
 # rotation-scheduler
```
Comment (dorbin) [inline README.md]: I know the actual repo name needs to be hyphenated, but in common usage could this be called "Rotation Scheduler"?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,96 @@
...
 A GitHub Action for generation a rotation schedule.
```
Comment (dorbin) [inline README.md]:
```suggestion
A GitHub Action to generate a rotation schedule.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,96 @@
...
+  user: cfieber
+- startDate: Sun 15 Mar 2020
+  user: ethanfrogers
+- startDate: Sun 22 Mar 2020
+  stopDate: Sat 28 Mar 2020
+  user: ezimanyi
+```
+
+Sync schedule to Google Calendar. If `user` fields are email addresses, the email address is added as an event attendee.
+The `--calendarID` **must** be a G-Suite user account dedicated for this purpose - all events are cleared during a sync!
```
Comment (dorbin) [inline README.md]:
```suggestion
The `--calendarID` **must** be a G Suite user account dedicated for this purpose - all events are cleared during a sync!
```

File: docs/rotation.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+rotation generates, extends, and syncs rotation schedules.
+
+### Options
+
+```
+  -h, --help            help for rotation
+  -r, --record string   Record the responses from external dependencies to the specified file. Used for external dependency testing.
+```
+
```
Comment (dorbin) [inline docs/rotation.md:39]: Maybe include an example?

File: docs/rotation.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+## rotation
+
+rotation generates, extends, and syncs rotation schedules.
```
Comment (dorbin) [inline docs/rotation.md]:
```suggestion
`rotation` generates, extends, and syncs rotation schedules.
```

File: docs/rotation.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+### Synopsis
+
+rotation generates, extends, and syncs rotation schedules.
```
Comment (dorbin) [inline docs/rotation.md]:
```suggestion
`rotation` generates, extends, and syncs rotation schedules.
```

File: docs/rotation.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+## rotation
```
Comment (dorbin) [inline docs/rotation.md:1]:
```suggestion
## `rotation`
```

File: docs/rotation.md (hunk 6 above)
Comment (dorbin) [reply docs/rotation.md]: Here you're saying the same thing as above. Maybe put usage here, with the whole command itself plus the options?

File: docs/rotation_calendar.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+## rotation calendar
+
+Shared calender manipulation functions
+
+### Synopsis
+
+Shared calender manipulation functions
```
Comment (dorbin) [inline docs/rotation_calendar.md]: Same thing here; no need to repeat this sentence.

File: docs/rotation_calendar_sync.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
+## rotation calendar sync
+
+Sync a schedule to a shared calendar
```
Comment (dorbin) [inline docs/rotation_calendar_sync.md]:
```suggestion
Sync a schedule to a shared calendar.
```

File: docs/rotation_calendar_sync.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+### Synopsis
+
+Sync a schedule to a shared calendar
```
Comment (dorbin) [inline docs/rotation_calendar_sync.md]: Repeated

File: docs/rotation_schedule.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
+## rotation schedule
+
+Schedule creation and extension functions
```
Comment (dorbin) [inline docs/rotation_schedule.md]:
```suggestion
Schedule creation and extension functions.
```

File: docs/rotation_schedule.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+### Synopsis
+
+Schedule creation and extension functions
```
Comment (dorbin) [inline docs/rotation_schedule.md]: Repeated

File: docs/rotation_schedule_extend.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
+## rotation schedule extend
+
+Extends a previously generated schedule
+
+### Synopsis
+
+Extends a previously generated schedule
```
Comment (dorbin) [inline docs/rotation_schedule_extend.md]: So, these files seem to be auto generated. (I'm looking at the h5 at the end of each of these. Does that mean the auto-gen functionality forces this format (h2 name of command, then sentence, then h3 synopsis + the same sentence)?

Comment (ttomsu) [reply docs/rotation_schedule_extend.md]: Yes, these are autogenerated. It looks like the [format is hardcoded](https://github.com/spf13/cobra/blob/master/doc/md_docs.go#L54), but to your point about repetition it look like I can provide a `Long` value which should reduce the duplication.

File: README.md (hunk 1 above)
Comment (ttomsu) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (ttomsu) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (ttomsu) [reply README.md]: Done.

File: docs/rotation.md (hunk 4 above)
Comment (ttomsu) [reply docs/rotation.md:39]: Done.

File: docs/rotation.md (hunk 5 above)
Comment (ttomsu) [reply docs/rotation.md]: Done.

File: docs/rotation_calendar_sync.md (hunk 9 above)
Comment (ttomsu) [reply docs/rotation_calendar_sync.md]: Done.

File: docs/rotation_schedule.md (hunk 11 above)
Comment (ttomsu) [reply docs/rotation_schedule.md]: Done.

File: docs/rotation_schedule.md (hunk 12 above)
Comment (ttomsu) [reply docs/rotation_schedule.md]: Ack - fixed.

**Review by ttomsu — COMMENTED** [review]
Added some longer doc language to help with the repetition. PTAL

File: README.md [hunk 14]
```diff
@@ -1,2 +1,96 @@
...
+  user: cfieber
+- startDate: Sun 15 Mar 2020
+  user: ethanfrogers
+- startDate: Sun 22 Mar 2020
+  stopDate: Sat 28 Mar 2020
+  user: ezimanyi
+```
+
+Sync schedule to Google Calendar. If `user` fields are email addresses, the email address is added as an event attendee.
+The `--calendarID` **must** be a G Suite user account dedicated for this purpose - all events are cleared during a sync!
```
Comment (dorbin) [inline README.md]:
```suggestion
The `--calendarID` **must** be a G Suite user account dedicated for this purpose. All events are cleared during a sync!
```

Comment (dorbin) [reply README.md]: Missed this before. What does it mean that all events are cleared during a sync?

File: cmd/root.go [hunk 15]
```diff
@@ -12,17 +12,42 @@ const (
...
 
 var (
-	rootCmd = &cobra.Command{
-		Use:          "rotation",
-		Short:        "rotation generates, extends, and syncs rotation schedules.",
+	RootCmd = &cobra.Command{
+		Use:   "rotation",
+		Short: "`rotation` generates, extends, and syncs rotation schedules.",
+		Long: `This tool generates and manipulates a human-readable and version-controllable 
+text file for an on-call, chore, or duty rotations among a set of individuals. Each shifts only
```
Comment (dorbin) [inline cmd/root.go]:
```suggestion
text file for an on-call, chore, or duty rotations among a set of individuals. Each shift only 
```

File: docs/rotation_schedule.md [hunk 16]
```diff
@@ -0,0 +1,32 @@
+## rotation schedule
+
+Schedule creation and extension functions.
+
+### Synopsis
+
+These options control the shift output, like providing the user list(or how to get a user list) and how long to make each shift. For GitHub team integration,users can be invited to their shift by making their email public.
```
Comment (dorbin) [inline docs/rotation_schedule.md]:
```suggestion
These options control the shift output, like providing the user list (or how to get a user list) and how long to make each shift. For GitHub team integration, users can be invited to their shift by making their email public.
```

File: docs/rotation_schedule_extend.md [hunk 17]
```diff
@@ -0,0 +1,41 @@
+## rotation schedule extend
+
+Extends a previously generated schedule.
+
+### Synopsis
+
+Once an initial schedule is in place, time will pass and that schedule
+needs to be extended further into the future. During that time, users sometimes get 
+added or removed from the rotation. By default, the previous scheduled shifts won't be modified.
+if --prune is true, however, all shifts are reviewed to ensure a current member of the rotation
```
Comment (dorbin) [inline docs/rotation_schedule_extend.md]:
```suggestion
if `--prune` is true, however, all shifts are reviewed to ensure a current member of the rotation
```

File: docs/rotation_schedule_generate.md [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+```
+rotation schedule generate [outputFile] [flags]
+```
+
+### Options
+
+```
+  -h, --help           help for generate
+      --start string   Optional. Generate schedule starting on this date (inclusive). Defaults to tomorrow's date. Must be in the format 2006-01-02
```
Comment (dorbin) [inline docs/rotation_schedule_generate.md]:
```suggestion
      --start string   Optional. Generate schedule starting on this date (inclusive). Defaults to tomorrow's date. Must be in the format yyyy-mm-dd
```

File: README.md (hunk 14 above)
Comment (ttomsu) [reply README.md]: All events on that users's calendar are wiped clean (deleted) first, and then the new schedule is put in place. It was the easiest sync logic I could come up with.

File: cmd/root.go (hunk 15 above)
Comment (ttomsu) [reply cmd/root.go]: Done.

File: docs/rotation_schedule.md (hunk 16 above)
Comment (ttomsu) [reply docs/rotation_schedule.md]: Done.

File: docs/rotation_schedule_generate.md (hunk 18 above)
Comment (ttomsu) [reply docs/rotation_schedule_generate.md]: Done.

File: docs/rotation_schedule_extend.md (hunk 17 above)
Comment (ttomsu) [reply docs/rotation_schedule_extend.md]: Done.

**Review by dorbin — APPROVED** [review]

**Pull request merged**